hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -702,15 +735,72 @@ public OutputStream openFileForWrite(final Path path, final FileSystem.Statistic
         isAppendBlob = true;
       }
 
+      SelfRenewingLease lease = maybeCreateLease(relativePath);
+
       return new AbfsOutputStream(
           client,
           statistics,
           relativePath,
           offset,
-          populateAbfsOutputStreamContext(isAppendBlob));
+          populateAbfsOutputStreamContext(isAppendBlob, lease));
     }
   }
 
+  /**
+   * Acquire a lease on an ABFS file for a specified duration. This requires the file to exist.
+   *
+   * @param path file name
+   * @param duration time lease will be held before expiring
+   * @return the acquired lease ID
+   * @throws AzureBlobFileSystemException on any exception while acquiring the lease
+   */
+  public String acquireLease(final Path path, final int duration) throws AzureBlobFileSystemException {
+    LOG.debug("lease path: {}", path);
+
+    final AbfsRestOperation op =
+        client.acquireLease(getRelativePath(path), duration);
+
+    return op.getResult().getResponseHeader(HttpHeaderConfigurations.X_MS_LEASE_ID);
+  }
+
+  /**
```

Comment:
this code should not be required

Reply:
I'm not sure what you mean. Do you mean that we shouldn't have an acquireLease method in the store because leases will be acquired automatically?

Reply:
Yes if the file is being created then infinite lease can be automatically taken, for other cases yes you may need the acquire lease code.

RenewLease code might be something you can completely get rid of